### PR TITLE
Ferigis.129449237.ignore blocks

### PR DESCRIPTION
**Pull request by ferigis**
In this PR there are two main features, "Ignore Blocks" and "include line number on the results".

### CODE REVIEW

File: src/sheldon_result.erl [hunk 1]
```diff
@@ -25,19 +25,27 @@
...
+             , misspelled_word/0
+             , line_number/0
              ]).
 
--type result() :: ok | #{ result  => [string()]
-                        , bazinga => string()
-                        }.
+-type line_number() :: non_neg_integer().
+-type misspelled_word() ::
+  #{ line_number := non_neg_integer()
```
Comment (gadgetci) [inline src/sheldon_result.erl]: According to **Xref**:
> syntax error before: ':='

File: src/sheldon_result.erl [hunk 2]
```diff
@@ -25,19 +25,27 @@
...
+  ok | #{ misspelled_words := [misspelled_word()]
+        , bazinga          := string()
+        }.
 
 %%%===================================================================
 %%% API
 %%%===================================================================
 
--spec result([string()], sheldon_config:config()) -> result().
+-spec result([misspelled_word()], sheldon_config:config()) -> result().
```
Comment (gadgetci) [inline src/sheldon_result.erl]: According to **Xref**:
> type result() undefined

File: src/sheldon_result.erl [hunk 3]
```diff
@@ -25,19 +25,27 @@
...
--type result() :: ok | #{ result  => [string()]
-                        , bazinga => string()
-                        }.
+-type line_number() :: non_neg_integer().
+-type misspelled_word() ::
+  #{ line_number := non_neg_integer()
+   , word        := string()
+   }.
+-type result() ::
+  ok | #{ misspelled_words := [misspelled_word()]
```
Comment (gadgetci) [inline src/sheldon_result.erl]: According to **Xref**:
> syntax error before: ':='

File: src/sheldon_result.erl (hunk 2 above)
Comment (gadgetci) [inline src/sheldon_result.erl]: According to **Xref**:
> type misspelled_word() undefined

**Comment by gadgetci** [comment]
**xref** failed with exit status: 1

File: src/sheldon_result.erl (hunk 1 above)
Comment (elbrujohalcon) [inline src/sheldon_result.erl]: shouldn't it be `line_number()` ?

File: src/sheldon_utils.erl [hunk 4]
```diff
@@ -31,29 +31,30 @@
...
-                , ")"
-                , "("
+                , "[?]"
+                , "[)]"
+                , "[(]"
                 , "\""
                 , "\'"
                 , "!"
-                , "["
+                , "\\["
```
Comment (elbrujohalcon) [inline src/sheldon_utils.erl]: you can use `[[]` here, too

File: src/sheldon_result.erl (hunk 1 above)
Comment (ferigis) [inline src/sheldon_result.erl]: yeap, I wrote that line_number() but I forgot to update the misspelled_word()...

File: test/sheldon_SUITE.erl [hunk 5]
```diff
@@ -44,33 +46,33 @@ end_per_suite(Config) ->
 -spec basic_check(config()) -> ok.
 basic_check(_Config) ->
   ok = sheldon:check("Hi, I am testing the spelling checker"),
-  #{result := ["wrongspelled"]} =
+  #{ misspelled_words := [#{ word := "wrongspelled" }] } =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/sheldon_SUITE.erl [hunk 6]
```diff
@@ -44,33 +46,33 @@ end_per_suite(Config) ->
...
     sheldon:check("I am testing a wrongspelled word"),
   ok = sheldon:check("Hi, I am (testing the spelling) checker"),
   ok = sheldon:check("Hi, I am \"testing the\" spelling checker"),
   ok = sheldon:check("Hi, I am \'testing\' the spelling checker"),
   ok = sheldon:check("Hi, I am \'testing\' the.,; spelling checker"),
-  #{result := ["thedfs"]} =
+  #{ misspelled_words := [#{ word := "thedfs" }] } =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/sheldon_SUITE.erl [hunk 7]
```diff
@@ -44,33 +46,33 @@ end_per_suite(Config) ->
...
     sheldon:check("Hi, I am \'testing\' thedfs.,; spelling checker"),
   ok = sheldon:check(""),
   ok = sheldon:check("I am checking numbers too 12, 34, 111, yeah!"),
   ok = sheldon:check("I am checking numbers too [12, 34], {111}, yeah!"),
   ok = sheldon:check("                                hello"),
-  #{result := ["Sheldon"]} =
+  #{ misspelled_words := [#{ word := "Sheldon" }] } =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/sheldon_SUITE.erl [hunk 8]
```diff
@@ -44,33 +46,33 @@ end_per_suite(Config) ->
...
     sheldon:check("Sheldon doesn't know his name, too bad"),
   ok = sheldon:check("I am checking numbers too [12, 34], {111}, yeah!", #{}),
   ok.
 
 -spec iodata_check(config()) -> ok.
 iodata_check(_Config) ->
-  #{result := ["iodata"]} =
+  #{misspelled_words := [#{ word := "iodata" }]} =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/sheldon_SUITE.erl [hunk 9]
```diff
@@ -44,33 +46,33 @@ end_per_suite(Config) ->
...
     sheldon:check(["I am",  [<<" testing with">>, " "] , <<"iodata">>]),
   ok = sheldon:check(["Hi, I am (testing the spelling) checker"]),
   ok = sheldon:check([ $H
                      , [<<"i">>, " I am"]
-                     , <<" (testing the spelling) checker">>
+                     , <<" (testing the spelling) checker !">>
                      ]),
-  #{result := ["thedfs"]} =
+  #{ misspelled_words := [#{ word := "thedfs" }] } =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/sheldon_SUITE.erl [hunk 10]
```diff
@@ -80,47 +82,141 @@ iodata_check(_Config) ->
...
 ignore_words(_Config) ->
-  Config = #{ignore_words => ["Sheldon"] },
+  Config = #{ ignore_words => ["Sheldon"] },
   ok = sheldon:check("Sheldon doesn't know his name, too bad", Config),
-  Config2 = #{ignore_words => ["ttttthis", "sheldon", "differentone"] },
+  Config2 = #{ ignore_words => ["ttttthis", "sheldon", "differentone"] },
   ok =
     sheldon:check(<<"testing again. Sheldon should skip ttttthis">>, Config2),
-  #{result := ["vvvvv"]} =
+  #{ misspelled_words := [#{ word := "vvvvv" }] } =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/sheldon_SUITE.erl [hunk 11]
```diff
@@ -80,47 +82,141 @@ iodata_check(_Config) ->
...
 -spec ignore_patterns(config()) -> ok.
 ignore_patterns(_Config) ->
-  Config = #{ignore_patterns => ["^begin"]},
+  Config = #{ ignore_patterns => ["^begin"] },
   ok = sheldon:check("begindddd should be ignored", Config),
-  Config2 = #{ignore_patterns => ["^[4-6]", "^begin"]},
+  Config2 = #{ ignore_patterns => ["^[4-6]", "^begin"] },
   ok = sheldon:check("begindddd and 4rrrrrr should be ignored", Config2),
-  #{result := ["7rrrr"]} =
+  #{ misspelled_words := [#{ word := "7rrrr" }] } =
```
Comment (harenson) [inline test/sheldon_SUITE.erl]: You're missing the match for the `line_number`.

File: test/files/multiline.txt [hunk 12]
```diff
@@ -0,0 +1,4 @@
+I want to check sheldon [ in a multiline Text
+it should show fsdfdsd as a wrong spelled word at line
+2. superwrong should appears at line 3.
```
Comment (elbrujohalcon) [inline test/files/multiline.txt]: should _appear_

**Pull request merged**